### PR TITLE
Wasm: Correctly implement `JSLinkingInfo`.

### PR DESCRIPTION
**Pull request by sjrd**
Review by @gzm0 and @tanishiking.

### CODE REVIEW

File: linker/shared/src/main/scala/org/scalajs/linker/backend/WebAssemblyLinkerBackend.scala [hunk 1]
```diff
@@ -32,6 +32,10 @@ import org.scalajs.linker.backend.wasmemitter.Emitter
 final class WebAssemblyLinkerBackend(config: LinkerBackendImpl.Config)
     extends LinkerBackendImpl(config) {
 
+  require(
+    coreSpec.esFeatures.esVersion >= ESVersion.ES2015,
+    s"The WebAssembly backend requires at least ES2015; was ${coreSpec.esFeatures.esVersion}."
+  )
```
Comment (gzm0) [inline linker/shared/src/main/scala/org/scalajs/linker/backend/WebAssemblyLinkerBackend.scala:38]: I don't understand how this check relates to this PR.

**Review by gzm0 — CHANGES_REQUESTED** [review]
Just one question.

File: linker/shared/src/main/scala/org/scalajs/linker/backend/WebAssemblyLinkerBackend.scala (hunk 1 above)
Comment (sjrd) [reply linker/shared/src/main/scala/org/scalajs/linker/backend/WebAssemblyLinkerBackend.scala]: Ah yes ... it's not. I thought it was at some point, but in the end that's a different thing. And it's not even about `esVersion` per se, but rather about `useECMAScript2015Semantics`. I'll make a separate PR later about that. I removed it from this one.

**Review by tanishiking — APPROVED** [review]
LGTM

**Review by gzm0 — APPROVED** [review]

**Pull request merged**